kdbg/dbgdriver.h

```diff
@@ -409,6 +412,13 @@ class DebuggerDriver : public QProcess
      */
     virtual void setPrintQStringDataCmd(const char* cmd) = 0;
 
+#ifdef Q_PROCESSOR_X86
+    /**
+     * Set the debugger's disassembly flavor
+     */
+    virtual void setDisassemblyFlavor(const QString& s) = 0;
+#endif
+
```

Comment:
I would prefer to have this function and the infrastructure that it needs unconditionally, i.e., without the `#ifdef Q_PROCESSOR_X86` guard.

Reply:
What should these functions do when ran for gdb on RPi for example?

Reply:
They should do the same as everywhere else *except* that the line that submits `DCdisassemblyFlavor` is inside a `#ifdef Q_PROCESSOR_X86` brackets so that it does not happen.

BTW, what happens when I debug an Intel process remotely from an RPi? Should I see Intel disassembly? Does that not mean that the disassembly flavor is process specific?

Reply:
In this case, you are right. The functions should exist, and I will make use of runtime checks (instead of compile-time). Right now I am more concerned on making the feature actually work though.

Reply:
I forgot to answer the last part. Yes, you will see Intel or att assembly. Like when you remotely debug a riscv executable, you will see riscv assembly. The assembly depends on the file to be debugged.